File: python-sdk/src/boxsdk/models/shared/workflows.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import workflow as shared_workflow


@dataclass_json
@dataclasses.dataclass
class Workflows:
    r"""Workflows
    The part of an API response that describes marker
    based pagination
    """
    
    entries: Optional[list[shared_workflow.Workflow]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('entries') }})
    limit: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('limit') }})
    next_marker: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next_marker') }})
    prev_marker: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('prev_marker') }})
    

File: ts-sdk/src/sdk/models/shared/workflows.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Workflow } from "./workflow";



// Workflows
/** 
 * The part of an API response that describes marker
 * based pagination
**/
export class Workflows extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=entries", elemType: Workflow })
  entries?: Workflow[];

  @SpeakeasyMetadata({ data: "json, name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "json, name=next_marker" })
  nextMarker?: number;

  @SpeakeasyMetadata({ data: "json, name=prev_marker" })
  prevMarker?: number;
}
